nodeup/pkg/model/kubelet.go

```diff
@@ -441,6 +442,23 @@ func (b *KubeletBuilder) buildKubeletConfigSpec() (*kops.KubeletConfigSpec, erro
 		c.BootstrapKubeconfig = ""
 	}
 
+	if b.Cluster.Spec.Networking != nil && b.Cluster.Spec.Networking.AmazonVPC != nil {
+		instanceType, err := awsup.GetMachineTypeInfo(b.InstanceGroup.Spec.MachineType)
+		if err != nil {
+			return c, err
+		}
+
+		maxPods := int32(instanceType.MaxPods)
+		c.MaxPods = &maxPods
+		if b.InstanceGroup.Spec.Kubelet != nil {
+			if b.InstanceGroup.Spec.Kubelet.MaxPods == nil {
+				b.InstanceGroup.Spec.Kubelet.MaxPods = &maxPods
```

Comment:
So I would use instance type maxPods as a maximum value here (I think).  My understanding is that it doesn't make any sense to set it any higher, because the AWS VPC CNI provider simply won't let any pods >= instanceType.maxPods work (?)